### PR TITLE
Fusion: Add damage reduction

### PR DESCRIPTION
**Pull request by Miepee**
Heat and Electric water damage are based off of the number of suits and thus can't afaik be accurately modelled right now (other than just having different values in logic).

### CODE REVIEW

File: randovania/games/fusion/generator/bootstrap.py [hunk 1]
```diff
@@ -35,6 +37,21 @@ def _get_enabled_misc_resources(
...
 
+    def _damage_reduction(self, db: ResourceDatabase, current_resources: ResourceCollection) -> float:
+        num_suits = sum(
+            (1 if current_resources[db.get_item_by_name(suit)] else 0) for suit in ("Varia Suit", "Gravity Suit")
+        )
+        dr = 1.0
+        if num_suits == 1:
+            dr = 0.9
+        elif num_suits >= 2:
+            dr = 0.9
```
Comment (JeffGainsNGames) [inline randovania/games/fusion/generator/bootstrap.py:48]: 0.8?

Comment (Miepee) [reply randovania/games/fusion/generator/bootstrap.py:48]: oop

File: randovania/games/fusion/generator/bootstrap.py [hunk 2]
```diff
@@ -35,6 +37,21 @@ def _get_enabled_misc_resources(
...
 
+    def _damage_reduction(self, db: ResourceDatabase, current_resources: ResourceCollection) -> float:
+        num_suits = sum(
+            (1 if current_resources[db.get_item_by_name(suit)] else 0) for suit in ("Varia Suit", "Gravity Suit")
+        )
+        dr = 1.0
+        if num_suits == 1:
+            dr = 0.9
+        elif num_suits >= 2:
+            dr = 0.8
```
Comment (JeffGainsNGames) [inline randovania/games/fusion/generator/bootstrap.py:48]: iirc this isnt correct. varia always gives the first set of DR and Grav gives the second. regardless of amount of suits. may need someone to confirm that.

Comment (Miepee) [reply randovania/games/fusion/generator/bootstrap.py:48]: Anty confirmed this a long time ago, that suit damage is progressive

Comment (Miepee) [reply randovania/games/fusion/generator/bootstrap.py:48]: > anyway enemy damage reduction can be 30%/10%, 60%/30%, 80%/10%, and 90%/80%
left being single suit, right being both

**Review by JeffGainsNGames — APPROVED** [review]

**Pull request merged**